db/table_cache.cc

```diff
@@ -336,10 +337,26 @@ Status TableCache::Get(const ReadOptions& options,
 
     if (auto row_handle =
             ioptions_.row_cache->Lookup(row_cache_key.GetUserKey())) {
+
+      // Cleanable routine to release the cache entry
+      Cleanable cache_entry_clean;
+      auto release_cache_entry_func = [] (void* cache_to_clean, void* cache_handle){
+          ((Cache*)cache_to_clean)->Release((Cache::Handle*) cache_handle);
+      };
       auto found_row_cache_entry = static_cast<const std::string*>(
           ioptions_.row_cache->Value(row_handle));
-      replayGetContextLog(*found_row_cache_entry, user_key, get_context);
-      ioptions_.row_cache->Release(row_handle);
+      // If it comes here value is located on the cache,
+      // found_row_cache_entry points to the value on cache,
+      // cache_entry_clean has cleanup procedure for the cached entry,
+      // after replayGetContextLog() returns  get_context.pinnable_slice_
+      // will point to cache entry buffer and
+      // cleanup routine under cache_entry_clean will be delegated to
+      // get_context.pinnable_slice_, cache entry is released when
+      // get_context.pinnable_slice_ is reset
+      cache_entry_clean.RegisterCleanup(release_cache_entry_func,
```

Comment:
Can you please apply this comment: https://github.com/facebook/rocksdb/pull/2492/commits/ee5273ebf4ba8e8b703b63a1069908beeb65693c#r124070427